codemark/signup.py

```diff
@@ -0,0 +1,112 @@
+import getpass
+import codemark.firebase.database as FireDB
+from codemark.utils import print_info, print_error, print_warning, print_success
+from codemark.submit import generate_hash
+
+db = FireDB.FirebaseDB()
+
+def signup():
+    print_info("Are you a student or a professor? ((s)tudent/(p)rofessor")
+    resp = input('Response: ')
+    
+    # if the user is a student
+    if resp == 's':
+        print_info('Enter your details to sign up!')
+        email = input('Enter email: ')
+        if not db.check_mail(email):
+            print_error('Invalid Email')
+            return False
+
+        password = getpass.getpass("Enter Password : ")
+        if len(password) < 8:
+            print_warning("Please enter at-least 8 character password!")
+            return False
+
+        student_id = generate_hash()
+        
+        name = input('Enter your name: ')
+        college_name = input('Enter the name of your college: ')
+        passing_year = input('Enter your year of passing: ')
+        field_of_study = input('Enter your field of study: ')
+        batch_semester = input('Enter your batch semester (1-8): ')
+        
+        if int(batch_semester) not in list(range(1, 9)):
```

Comment:
This will possibly  error out in case batch_semester got string values. Ensure proper checking for the same.